Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,329 @@
-﻿namespace BootCamp.Chapter
+﻿using System.Text;
+
+namespace BootCamp.Chapter
 {
     public static class BalanceStats
     {
+        private const string invalidMessage = "N/A.";
+        private const string currencySymbol = "¤";
+
         /// <summary>
         /// Return name and balance(current) of person who had the biggest historic balance.
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+
+            float highestBalanceEver = GetHighestBalanceEver(peopleAndBalances);
+            string[] highestBalanceNames = FindBalanceNames(highestBalanceEver, peopleAndBalances, false);
+
+            return $"{highestBalanceNames[0]} had the most money ever. {currencySymbol}{highestBalanceEver}.";
         }
 
         /// <summary>
         /// Return name and loss of a person with a biggest loss (balance change negative).
         /// </summary>
         public static string FindPersonWithBiggestLoss(string[] peopleAndBalances)
         {
-            return "";
+            if (IsArrayNullOrEmpty(peopleAndBalances))
+            {
+                return invalidMessage;
+            }
+            string[][] workingPeopleAndBalances = ReturnJaggedArray(peopleAndBalances);
```

Comment:
So what that you return a jagged array? Is it hardcoded? Is it made of something? PLease rename better,. For  example: ConvertToJaggedArray is a way better name.